src/benchmarklib/abstract_table_generator.cpp

```diff
@@ -456,21 +456,31 @@ bool AbstractTableGenerator::_all_chunks_sorted_by(const std::shared_ptr<Table>&
 std::unordered_map<std::string, BenchmarkTableInfo> AbstractTableGenerator::_load_binary_tables_from_path(
     const std::string& cache_directory) {
   auto table_info_by_name = std::unordered_map<std::string, BenchmarkTableInfo>{};
+  const auto table_files = list_directory(cache_directory);
+  auto jobs = std::vector<std::shared_ptr<AbstractTask>>{};
+  jobs.reserve(table_files.size());
 
-  for (const auto& table_file : list_directory(cache_directory)) {
+  for (const auto& table_file : table_files) {
     const auto table_name = table_file.stem();
-    std::cout << "-  Loading table '" << table_name.string() << "' from cached binary " << table_file.relative_path();
+    table_info_by_name[table_name] = BenchmarkTableInfo{};
 
-    auto timer = Timer{};
-    auto table_info = BenchmarkTableInfo{};
-    table_info.table = BinaryParser::parse(table_file);
-    table_info.loaded_from_binary = true;
-    table_info.binary_file_path = table_file;
-    table_info_by_name[table_name] = table_info;
+    jobs.emplace_back(std::make_shared<JobTask>([table_name, table_file, &table_info_by_name]() {
+      auto message = std::stringstream{};
+      message << "-  Loaded table '" << table_name.string() << "' from cached binary " << table_file.relative_path();
 
-    std::cout << " (" << timer.lap_formatted() << ")\n";
+      auto timer = Timer{};
+      auto& table_info = table_info_by_name[table_name];
+      table_info.table = BinaryParser::parse(table_file);
+      table_info.loaded_from_binary = true;
+      table_info.binary_file_path = table_file;
+
+      message << " (" << timer.lap_formatted() << ")\n";
+      std::cout << message.str();
```

Comment:
Can we flush the lines? I just checked SF 100 for the other data loading PR and often (if not all the time) all import message are coming altogether. Especially for large SFs, it would be helpful to get them immediately.